libsigopt/aux/validate_schema.py

```diff
@@ -0,0 +1,81 @@
+# Copyright © 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache License 2.0
+
+import json
+import re
+from collections.abc import Mapping
+from numbers import Integral
+
+from jsonschema import validate as validate_against_schema  # type: ignore
+from jsonschema.exceptions import ValidationError  # type: ignore
+
+from libsigopt.aux.errors import (
+  InvalidKeyError,
+  InvalidTypeError,
+  InvalidValueError,
+  MissingJsonKeyError,
+  SigoptValidationError,
+)
+
+
+def validate(json_dict, schema) -> None:
+  try:
+    validate_against_schema(json_dict, schema)
+  except ValidationError as e:
+    raise process_error(e) from e
+
+
+def is_integer(num):
+  if isinstance(num, bool):
+    return False
+  elif isinstance(num, Integral):
```

Comment:
`isinstance(num, int)` is probably also valid.

Reply:
Is this used in any numpy checking stuff (and does Integral handle that correctly, whereas int doesn't)?